ant-node/tests/simulate_replication.rs

```diff
@@ -0,0 +1,1373 @@
+    let avg_estimate = total_estimate as f64 / nodes.len() as f64;
+    let min_estimate = nodes
+        .values()
+        .map(|n| n.network_size_estimate)
+        .min()
+        .unwrap_or(0);
+    let max_estimate = nodes
+        .values()
+        .map(|n| n.network_size_estimate)
+        .max()
+        .unwrap_or(0);
+    println!(
+        "  ✓ Network size estimates: avg {avg_estimate:.0}, min {min_estimate}, max {max_estimate} (actual: {num_nodes}) - {:.2}s",
+        estimation_start.elapsed().as_secs_f64()
+    );
+
+    // Calculate responsible distances in parallel
+    println!("Calculating responsible distances...");
+    let resp_dist_start = std::time::Instant::now();
+    nodes.par_iter_mut().for_each(|(_, node)| {
+        node.calculate_responsible_distance();
+    });
+    println!(
+        "  ✓ Calculated responsible distances - {:.2}s",
+        resp_dist_start.elapsed().as_secs_f64()
+    );
+    println!(
+        "Phase 1 complete - {:.2}s\n",
+        phase1_start.elapsed().as_secs_f64()
+    );
+
+    // ========================================================================
+    // Phase 2: Client Upload
+    // ========================================================================
+    println!("Phase 2: Uploading {num_chunks} chunks with {payment_mode:?} payment mode...");
+    let phase2_start = std::time::Instant::now();
+
+    let chunk_addresses = Mutex::new(Vec::new());
+    let upload_stats = Mutex::new(UploadStats::new());
+    let nodes = Mutex::new(nodes);
+    let uploaded_count = AtomicUsize::new(0);
+
+    (0..num_chunks).into_par_iter().for_each(|_| {
+        // Generate random chunk
+        let chunk_addr = NetworkAddress::ChunkAddress(ChunkAddress::new(XorName::random(
+            &mut rand::thread_rng(),
+        )));
+        let chunk_size = 1024 * 1024; // 1MB
+
+        // Add to chunk_addresses (thread-safe)
+        chunk_addresses.lock().unwrap().push(chunk_addr.clone());
+
+        // Find 5 closest nodes globally (perfect client)
+        let mut closest_5: Vec<_> = all_peer_ids
+            .iter()
+            .map(|peer| {
+                let addr = NetworkAddress::from(*peer);
+                (*peer, chunk_addr.distance(&addr))
+            })
+            .collect();
+        closest_5.sort_by_key(|(_, dist)| *dist);
+        let closest_5: Vec<_> = closest_5.into_iter().take(CLOSE_GROUP_SIZE).collect();
+
+        // Get quotes from all 5 (needs read access to nodes)
+        let mut quotes: Vec<_> = {
+            let nodes_guard = nodes.lock().unwrap();
+            closest_5
+                .iter()
+                .map(|(peer, _)| (*peer, nodes_guard[peer].generate_quote(chunk_size)))
+                .collect()
+        };
+        quotes.sort_by_key(|(_, price)| *price);
+
+        // Select payment targets based on mode
+        let paid_nodes: Vec<PeerId> = match payment_mode {
+            PaymentMode::SingleNode => {
```

Comment:
all candidates that got picked in the payment, even they got 0 paid, shall still accept the upload.
hence here, all 5 shall be included in the payment. otherwise will result in rejection of `NotInPaymentList`